xv6-public/sysproc.c

```diff
@@ -6,6 +6,34 @@
 #include "memlayout.h"
 #include "mmu.h"
 #include "proc.h"
+#include "spinlock.h"
+
+struct {
+  struct spinlock lock;
+  struct proc proc[NPROC];
+} ptable;
+		
```

Comment:
Instead of adding this structure here, maybe put a function in proc.c to return the ptable